File: 7-reverse-integer/reverse-integer.cs
public class Solution {
    public int Reverse(int x) {
        string b = x.ToString();
        string c = "";
        string d = "";
        int f = 0;
        if(b[0]=='-')
        {
            c = b.Substring(1);
        }
        else
        {
            c = b;
        }
        for(int i = c.Length-1; i >= 0; i--)
        {
            d += c[i];
        }
        long e = long.Parse(d);
        if(x<0)
        {
            e = -e;
        }
        if(e >= int.MinValue && e <= int.MaxValue-1)
        {
            f = (int)e;
            return f;
        }
        else
        {
            return 0;
        }
    }
}

File: 7-reverse-integer/reverse-integer.py
class Solution(object):
    def reverse(self, x):
        """
        :type x: int
        :rtype: int
        """
        x = str(x)
        if x[0]=='-':
            p = int(x[1:][::-1])
            p=-p
        else:
            p=int(x[::-1])
        if p>(-2)**31 and p<(2**31)-1:
            return p
        return 0